editor/src/components/canvas/controls/comment-indicator.tsx

```diff
@@ -32,98 +41,185 @@ export const CommentIndicator = React.memo(() => {
+      onMouseDown={onMouseDown}
+    >
+      <div
+        css={{
+          height: 24,
+          width: 24,
+          background: 'black',
+          border: '1px solid ',
+          borderRadius: '24px 24px 24px 0px',
+          display: 'flex',
+          alignItems: 'center',
+          justifyContent: 'center',
+        }}
+      >
+        <div
+          style={{
+            height: 20,
+            width: 20,
+            borderRadius: 10,
+            background: color.background,
+            color: color.foreground,
+            fontSize: 9,
+            fontWeight: 'bold',
+            display: 'flex',
+            alignItems: 'center',
+            justifyContent: 'center',
+            boxShadow: UtopiaTheme.panelStyles.shadows.medium,
+          }}
+        >
+          <AvatarPicture url={avatar} initials={initials} />
+        </div>
+      </div>
+    </div>
   )
 })
-CommentIndicatorInner.displayName = 'CommentIndicatorInner'
+CommentIndicator.displayName = 'CommentIndicator'
+
+function useDragging(thread: ThreadData<ThreadMetadata>) {
+  const editThreadMetadata = useEditThreadMetadata()
+  const dispatch = useDispatch()
+  const [dragPosition, setDragPosition] = React.useState<CanvasPoint | null>(null)
+
+  const canvasScaleRef = useRefEditorState((store) => store.editor.canvas.scale)
+  const canvasOffsetRef = useRefEditorState((store) => store.editor.canvas.realCanvasOffset)
+
+  const onMouseMove = React.useCallback(
+    (event: MouseEvent) => {
+      event.stopPropagation()
+      const newCanvasPoint = windowPositionToCanvasPosition(
+        canvasScaleRef.current,
+        canvasOffsetRef.current,
+        windowPoint({ x: event.clientX, y: event.clientY }),
+      )
+      setDragPosition(newCanvasPoint)
+    },
+    [canvasOffsetRef, canvasScaleRef],
+  )
+
+  const onMouseUp = React.useCallback(
+    (event: MouseEvent) => {
+      event.stopPropagation()
+      const newCanvasPosition = windowPositionToCanvasPosition(
+        canvasScaleRef.current,
+        canvasOffsetRef.current,
+        windowPoint({ x: event.clientX, y: event.clientY }),
+      )
+      setDragPosition(null)
+
+      editThreadMetadata({
+        threadId: thread.id,
+        metadata: {
+          x: newCanvasPosition.x,
+          y: newCanvasPosition.y,
+        },
+      })
+      window.removeEventListener('mousemove', onMouseMove)
+      window.removeEventListener('mouseup', onMouseUp)
+    },
+    [onMouseMove, canvasOffsetRef, canvasScaleRef, editThreadMetadata, thread.id],
+  )
+
+  const onMouseDown = React.useCallback(
+    (event: React.MouseEvent) => {
+      event.stopPropagation()
+      dispatch([switchEditorMode(EditorModes.commentMode(null, true))])
+      window.addEventListener('mousemove', onMouseMove)
+      window.addEventListener('mouseup', onMouseUp)
+    },
+    [onMouseMove, onMouseUp, dispatch],
+  )
+
+  return { onMouseDown, dragPosition }
+}
+
+function windowPositionToCanvasPosition(
```

Comment:
This doesn't feel like it should be in here and I'm a bit suspicious of those `-10` values!

Reply:
It is not a general function, it is just about offsetting the position so the center of the indicator is at the mouse not the top/left corner. I cleaned this up a bit in fb021103202f6e1734952b497dc1a0c7eacb2026